src/scripts/itembynamedetailpage.js

```diff
@@ -190,57 +248,126 @@ define(["connectionManager", "listView", "cardBuilder", "imageLoader", "libraryB
+                itemsContainer.classList.add("vertical-wrap");
+                itemsContainer.classList.remove("vertical-list");
+            }
+
+            itemsContainer.innerHTML = html;
+            imageLoader.lazyChildren(itemsContainer);
+        });
     }
 
     function getMoreItemsHref(item, type) {
-        return "Genre" == item.Type ? "list.html?type=" + type + "&genreId=" + item.Id + "&serverId=" + item.ServerId : "MusicGenre" == item.Type ? "list.html?type=" + type + "&musicGenreId=" + item.Id + "&serverId=" + item.ServerId : "Studio" == item.Type ? "list.html?type=" + type + "&studioId=" + item.Id + "&serverId=" + item.ServerId : "MusicArtist" == item.Type ? "list.html?type=" + type + "&artistId=" + item.Id + "&serverId=" + item.ServerId : "Person" == item.Type ? "list.html?type=" + type + "&personId=" + item.Id + "&serverId=" + item.ServerId : "list.html?type=" + type + "&parentId=" + item.Id + "&serverId=" + item.ServerId
+        if ("Genre" == item.Type) {
+            return "list.html?type=" + type + "&genreId=" + item.Id + "&serverId=" + item.ServerId;
+        }
+
+        if ("MusicGenre" == item.Type) {
+            return "list.html?type=" + type + "&musicGenreId=" + item.Id + "&serverId=" + item.ServerId;
+        }
+
+        if ("Studio" == item.Type) {
+            return "list.html?type=" + type + "&studioId=" + item.Id + "&serverId=" + item.ServerId;
+        }
+
+        if ("MusicArtist" == item.Type) {
+            return "list.html?type=" + type + "&artistId=" + item.Id + "&serverId=" + item.ServerId;
+        }
+
+        if ("Person" == item.Type) {
+            return "list.html?type=" + type + "&personId=" + item.Id + "&serverId=" + item.ServerId;
+        }
+
+        return "list.html?type=" + type + "&parentId=" + item.Id + "&serverId=" + item.ServerId;
     }
 
     function addCurrentItemToQuery(query, item) {
-        "Person" == item.Type ? query.PersonIds = item.Id : "Genre" == item.Type ? query.GenreIds = item.Id : "MusicGenre" == item.Type ? query.GenreIds = item.Id : "Studio" == item.Type ? query.StudioIds = item.Id : "MusicArtist" == item.Type && (connectionManager.getApiClient(item.ServerId).isMinServerVersion("3.4.1.18") ? query.AlbumArtistIds = item.Id : query.ArtistIds = item.Id)
+        if (item.Type == "Person") {
+            query.PersonIds = item.Id;
+        } else if (item.Type == "Genre") {
+            query.Genres = item.Name;
+        } else if (item.Type == "MusicGenre") {
+            query.Genres = item.Name;
+        } else if (item.Type == "GameGenre") {
+            query.Genres = item.Name;
+        } else if (item.Type == "Studio") {
+            query.StudioIds = item.Id;
+        } else if (item.Type == "MusicArtist") {
+            if (connectionManager.getApiClient(item.ServerId).isMinServerVersion("3.4.1.18")) {
+                query.AlbumArtistIds = item.Id;
+            } else {
+                query.ArtistIds = item.Id;
+            }
```

Comment:
We're on version 3.5 anyway, so let's simplify it to
```js
query.AlbumArtistIds = item.Id;
```

Reply:
https://github.com/jellyfin/jellyfin-web/pull/506/commits/03a16287a033c45812fe5ecee444d1b98fb207c3